core/model/src/main/java/org/eclipse/rdf4j/model/Triple.java

```diff
@@ -13,6 +13,9 @@
 /**
  * An RDF* triple. Triples have a subject, predicate and object. Unlike {@link Statement}, a triple never has an
  * associated context.
+ * <p>
+ * <strong>Warning</strong> / In order to ensure interoperability of concrete classes implementing this interface,
```

Comment:
This is a matter of taste but can we take off the `<strong>Warning</strong>` bit? It's important for people implementing the interface, but most people looking at the javadoc will just be interested in using it. Perhaps turn it into an `@implNote` entry.

Reply:
`@implNote` sounds indeed more correct: updated javadocs to this end.

Reply:
The same warning is present also on BNode/IRI/Literal in #2538 and it would be better to keep things aligned: where should we address this? new PR?

Reply:
@knoan sorry, I missed that. Yeah, separate PR I think since those changes are only on the develop branch (and this is a quick fix for the master branch). Not massively urgent though.